drivers/ethernet/eth_tsn_nic_tsn.c

```diff
@@ -268,19 +345,142 @@ static void bake_qos_config(struct tsn_config *config)
 static bool get_timestamps(struct timestamps *timestamps, const struct tsn_config *tsn_config,
 			   net_time_t from, uint8_t tc_id, uint64_t bytes, bool consider_delay)
 {
-	/* TODO: Implement Qbv */
-	/* FIXME: These are for suppressing compile warnings */
-	timestamps->from = 0;
-	timestamps->to = 0;
-	timestamps->delay_from = 0;
-	timestamps->delay_to = 0;
-	return false;
+	/**
+	 * NOTE: This is the exact same copy of the one in the Linux driver
+	 *       Review this when something unexpected happens
+	 */
+	int slot_id, slot_count;
+	net_time_t sending_duration, remainder;
+	const struct qbv_baked_config *baked;
+	const struct qbv_baked_prio *baked_prio;
+	const struct qbv_config *qbv = &tsn_config->qbv;
+
+	memset(timestamps, 0, sizeof(struct timestamps));
+
+	if (qbv->enabled == false) {
+		/* No Qbv. Just return the current time */
+		timestamps->from = from;
+		timestamps->to = TSN_ALWAYS_OPEN(timestamps->from);
+		/* delay_* is pointless. Just set it to be right next to the frame */
+		timestamps->delay_from = timestamps->from;
```

Comment:
바로 다음 프레임이라면 이게 맞지 않나요?

```suggestion
		/* delay_* is pointless. Just set it to be right next to the frame */
		timestamps->delay_from = timestamps->to;
```

Reply:
리눅스 드라이버에도 이렇게 되어 있는데 `to == from - 1`이라 `delay_from == to + 1`을 의도하고 이렇게 했던 걸로 기억합니다. 그런데 이게 더 직관적인 것 같네요. 수정했습니다.